### ISSUE TITLE
L059 not accounting for certain keywords

### ISSUE THREAD

**Issue by timcosta** [sqlfluff/sqlfluff#3746]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

It seems like L059 isn't recognizing certain reserved/unreserved keywords. I believe it has to do with the ones coming from the ansii base set of keywords, likely due to [this line or something close](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/rules/L059.py#L172). 

### Expected Behaviour

Given a query like
```sql
SELECT
    n AS name,
    t AS type,
    o AS owner
FROM table
```

and rules configured like
```ini
# when keywords can be used as identifiers
[sqlfluff:rules:L029]
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none
```

I would expect L029 to highlight `name`, `type`, and `owner` as needing quotes due to being keywords, which is what happens.

I would then expect L059 to **not** highlight the quoted values as they are identifiers that must be quoted. L059 is using default configurations.

### Observed Behaviour

L029 highlights the unquoted keywords with `Keywords should not be used as identifiers.`, which is correct.

L059 highlights them as `Unnecessary quoted identifier "name".`, which is **incorrect** because `name` is a keyword and should be exempted from the unquoted identifier requirements.

### How to reproduce

See expected behavior for `.sqlfluff` rule config and a sample query.

### Dialect

`athena`

### Version

```bash
$ sqlfluff --version
sqlfluff, version 1.2.1
```

### Configuration

```ini
[sqlfluff]
templater = placeholder
dialect = athena

rules = all
# excluding L026 and L028 because of high false positive rates in hive-like
# dialects, and athena is one because it supports nested structs
exclude_rules = L026, L028
ignore = templating

[sqlfluff:indentation]
indented_joins = True
indented_ctes = True

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 80
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = unqualified
unquoted_identifiers_policy = aliases

# keyword capitalization
[sqlfluff:rules:L010]
capitalisation_policy = upper

# non-keyword capitalization (fields, table names, etc)
[sqlfluff:rules:L014]
extended_capitalisation_policy = lower

# when keywords can be used as identifiers
[sqlfluff:rules:L029]
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by timcosta** [comment]
I may have time to submit a PR some time this week or next, but no promises.

**Comment by tunetheweb** [comment]
This is working as intended.

These are unreserved keywords. They CAN be used as aliases and do not need to be quoted to be used (as they are unreserved). Absolute best practice would be not to use even unreserved keywords, though personally I find that a bit restrictive.

So L029 is correct to flag them.
And L059 is correct to subsequently flag them.

In effect the rules are incompatible if you want to use unreserved words as aliases. L029 tries to stop you doing that, and L059 will not allow you to quote to get around that (because you do not need the quotes as they are unreserved!).

So, your choices are, one or these:
1) Stop using unreserved words as aliases.
2) Turn L029 off so these won't be flagged. Note that reserved words will still be flagged as it's not a parsing error. Personally that's what I do as find this rule pretty pointless and annoying.
3) Allow these three words for L029 by using the `ignore_words` option L029 but not allowing any unreserved keywords to be used, if you can't/don't want to do option 1) or 2) but want to limit the issue to a small subset of "allowed" unreserved words.
4) Use the unnecessary quotes and turn off L059 completely.
5) Use the unnecessary quotes and turn off L059 just for these three unreserved keywords with ignore words option.

Personally my choice is 2).

**Comment by timcosta** [comment]
interesting, thanks for the feedback. we unfortunately have to have some aliases using those words, so i'll have to explicitly ignore them or disable L029 entirely. i appreciate the super detailed response, and am sure many folks in the future will too!
